### PR TITLE
Fix: Ignore anonymous classes

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] asserts that anonymous classes are ignored
* [x] ignores anonymous classes

Follows #29.

### CODE REVIEW

**Comment by weierophinney** [comment]
@localheinz Please rebase to resolve conflicts, which will trigger a new build as well. Thanks!

**Comment by localheinz** [comment]
@weierophinney 

Sorted!

**Comment by localheinz** [comment]
And thanks a lot for reviewing these on such a short notice, I appreciate it!

Since https://github.com/zendframework/modules.zendframework.com (where I am a maintainer) is kind of dead, I wouldn't mind getting involved as a maintainer with a different Zend project, please let me know if there's interest!

File: src/ClassFileLocator.php
Comment (localheinz) [inline src/ClassFileLocator.php]: Ha, should this be `$i > 1`, actually?

Comment (weierophinney) [reply src/ClassFileLocator.php]: You tell me; haven't played with tokenizing anonymous classes yet. :) What happens in the tests?

File: src/ClassFileLocator.php
Comment (weierophinney) [inline src/ClassFileLocator.php]: Use yoda conditions for the constant comparisons here.

File: test/TestAsset/Anonymous/WithAnonymousClass.php
Comment (weierophinney) [inline test/TestAsset/Anonymous/WithAnonymousClass.php]: Three changes:

- Reference the zend-file repository, using https.
- Since this file is just being introduced, change the copyright date to simply `2017`.
- Point to https://github.com/zendframework/zend-file/blob/master/LICENSE.md for the license.

File: test/TestAsset/Anonymous/with-anonymous-class.php
Comment (weierophinney) [inline test/TestAsset/Anonymous/with-anonymous-class.php]: Same notes on this docblock as for the previous.

**Review by weierophinney — CHANGES_REQUESTED** [review]
Noted a few changes. Additionally, in the files changed (but not introduced), update the copyright in the file-level docblock to reference 2017.

Thanks!

File: src/ClassFileLocator.php
Comment (localheinz) [inline src/ClassFileLocator.php]: Modified this condition

```diff
-&& $i > 2
+&& $i >+ 2
```

as the index doesn't need to be greater than 2, but at least 2 (since we try to access the element at `$i - 2`).

**Review by weierophinney — APPROVED** [review]

**Comment by weierophinney** [comment]
Will merge once I verify the build with Travis. Currently their running on a bit of a delay, due to the queue.

**Comment by localheinz** [comment]
Thanks a ton, @weierophinney!

**Comment by weierophinney** [comment]
@localheinz Already seeing errors on all jobs run so far: https://travis-ci.org/zendframework/zend-file/builds/190986651

File: test/TestAsset/Anonymous/with-anonymous-class.php
Comment (localheinz) [inline test/TestAsset/Anonymous/with-anonymous-class.php:11]: The question is, on the other hand, should the `ClassFileLocator` actually accept a PHP file that doesn't contain any classes?

Comment (weierophinney) [reply test/TestAsset/Anonymous/with-anonymous-class.php:11]: I would expect that it would ignore it.

Comment (weierophinney) [reply test/TestAsset/Anonymous/with-anonymous-class.php:11]: To clarify: the intent of the class is to find files that declare _named_ classes. It was originally developed as a utility for creating class maps for autoloading. Anonymous classes, as they are not named, would fall outside that criteria.

Comment (localheinz) [reply test/TestAsset/Anonymous/with-anonymous-class.php:11]: Will update later, @weierophinney!

**Comment by localheinz** [comment]
@weierophinney 

Amended the previous commit:

```diff
diff --git a/src/ClassFileLocator.php b/src/ClassFileLocator.php
index 82619a2..767fee4 100644
--- a/src/ClassFileLocator.php
+++ b/src/ClassFileLocator.php
@@ -124,8 +124,7 @@ class ClassFileLocator extends FilterIterator
                     }

                     // ignore anonymous classes on PHP 7.1 and greater
-                    if (PHP_VERSION_ID >= 70100
-                        && $i >= 2
+                    if ($i >= 2
                         && \is_array($tokens[$i - 1])
                         && T_WHITESPACE === $tokens[$i - 1][0]
                         && \is_array($tokens[$i - 2])
```

👍

File: src/ClassFileLocator.php
Comment (weierophinney) [inline src/ClassFileLocator.php]: Um... `&&` to start the conditional...? Also, will `$i >= 2` only ever happen on PHP 7.1, or will this work on any version?

**Review by weierophinney — CHANGES_REQUESTED** [review]

File: src/ClassFileLocator.php
Comment (localheinz) [reply src/ClassFileLocator.php]: Ha, hehe!

Well, I guess trying to run 

```php
$tokens   = token_get_all($contents);
```

when `$contents` contains usage of anonymous classes will already fail on a PHP version less than PHP 7.1, what do you think?

Comment (localheinz) [reply src/ClassFileLocator.php]: Actually, no, works fine on all PHP versions, see https://3v4l.org/0enag!

**Review by weierophinney — APPROVED** [review]

**Comment by localheinz** [comment]
Thanks a ton, @weierophinney!

**Pull request merged**
